docs/utilities/data_classes.md

```diff
@@ -679,7 +679,12 @@ Data classes and utility functions to help create continuous delivery pipelines
             else:
                 template = event.get_artifact(artifact_name, template_file)
                 # Kick off a stack update or create
-                start_update_or_create(job_id, stack, template)
+                result = start_update_or_create(job_id, stack, template)
+                event.put_artifact(
+                    artifact_name="json-artifact",
```

Comment:
Would it make more sense to dynamically call the artifact_name here? Like `event.data.input_artifacts[0].name` or `event.data.output_artifacts[0].name`

Reply:
That works well if you only have one artifact or if you know which order your artifacts returned are in, but to me it feels like an anti-pattern as your artifacts are defined in CodePipeline stage and are named.

Each stage in CodePipeline can have up to 5 artifacts.

Reply:
Thinking more about this, it's likely this is a more common convention and aligns more with the docs as a whole. I'll update it.

Reply:
Pushed an update to the docs.